### PR TITLE
Improved Exception Handling for Sniping. Cleaned and Optimized Sniping Code.

### PR DESCRIPTION
**Pull request by Daylend**
Lots of work done here. Timeout exceptions are now being handled, a lot of the code has been cleaned, removed a second call to search for pokemon on every snipe, renamed a config option (now called "UsePokeSnipersLocationServer") to make more sense, and a useragent is now being used so we don't get yelled at.

Code has been compiled + tested.

### CODE REVIEW

File: PoGo.PokeMobBot.Logic/Tasks/SnipePokemonTask.cs [hunk 1]
```diff
@@ -345,110 +344,79 @@ private static ScanResult SnipeScanForPokemon(ISession session, Location locatio
...
-                        });
                     }
-                    else if (resp.StatusCode == HttpStatusCode.BadGateway)
+                else if(ex.Status == WebExceptionStatus.Timeout)
                     {
-                        session.EventDispatcher.Send(new WarnEvent
-                        {
-                            Message = session.Translation.GetTranslation(TranslationString.WebErrorBadGateway)
-                        });
+                    session.EventDispatcher.Send(new WarnEvent { Message = "SkipLagged is down or SnipeRequestTimeoutSeconds is too small!" });
```
Comment (DurtyFree) [inline PoGo.PokeMobBot.Logic/Tasks/SnipePokemonTask.cs]: Please add a translation for that

File: PoGo.PokeMobBot.Logic/Tasks/SnipePokemonTask.cs [hunk 2]
```diff
@@ -523,8 +487,7 @@ private static ScanResult OnlineSnipeScanForPokemon(ISession session, Location l
                     if (offset < 0.001) offset = 0.003;
                     if (offset > 0.06) offset = 0.06;
 
-                    var uri =
-                        $"http://pokesnipers.com/api/v1/pokemon.json";
+                    var uri = $"http://pokesnipers.com/api/v1/pokemon.json";
```
Comment (DurtyFree) [inline PoGo.PokeMobBot.Logic/Tasks/SnipePokemonTask.cs]: Seems like you are using this quite often, consider doing a const variable for easy changing for that

File: PoGo.PokeMobBot.Logic/Tasks/SnipePokemonTask.cs [hunk 3]
```diff
@@ -564,71 +527,41 @@ private static ScanResult OnlineSnipeScanForPokemon(ISession session, Location l
...
-                                });
                             }
-                            else if (resp.StatusCode == HttpStatusCode.BadGateway)
-                            {
-                                session.EventDispatcher.Send(new WarnEvent
+                        else if (ex.Status == WebExceptionStatus.Timeout)
                                 {
-                                    Message = session.Translation.GetTranslation(TranslationString.WebErrorBadGateway)
-                                });
+                            session.EventDispatcher.Send(new WarnEvent { Message = "SkipLagged is down or SnipeRequestTimeoutSeconds is too small!" });
```
Comment (DurtyFree) [inline PoGo.PokeMobBot.Logic/Tasks/SnipePokemonTask.cs]: Wouldn't it be cool to have a translation string for that? :D jokes

**Comment by DurtyFree** [comment]
Please have a look at my comments, will merge then! Thx:)

File: PoGo.PokeMobBot.Logic/Tasks/SnipePokemonTask.cs (hunk 2 above)
Comment (Daylend) [inline PoGo.PokeMobBot.Logic/Tasks/SnipePokemonTask.cs]: Someone else's work. I'll add a const.

**Comment by Daylend** [comment]
Meeeeeeerge meeeeee

**Comment by DurtyFree** [comment]
Looks fine!:)

**Comment by vfuyong** [comment]
RELEASE!! ;)

**Pull request merged**